src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1797,15 +1797,36 @@ private function registerMessengerConfiguration(array $config, ContainerBuilder
             $container->setAlias('messenger.default_serializer', $config['serializer']['default_serializer']);
         }
 
+        $failureTransports = [];
+        if ($config['failure_transport']) {
+            if (!isset($config['transports'][$config['failure_transport']])) {
+                throw new LogicException(sprintf('Invalid Messenger configuration: the failure transport "%s" is not a valid transport or service id.', $config['failure_transport']));
+            }
+
+            $container->setAlias('messenger.failure_transports.default', $config['failure_transport']);
+            $failureTransports[] = $config['failure_transport'];
+        }
+        foreach ($config['transports'] as $name => $transport) {
+            if (isset($transport['failure_transport']) && !isset($config['transports'][$transport['failure_transport']])) {
+                throw new LogicException(sprintf('Invalid Messenger configuration: the failure transport "%s" is not a valid transport or service id.', $transport['failure_transport']));
```

Comment:
I think we can remove this LogicException and the one above it. We already have this check down below around line 1860. And, it's not *completely* correct up here, because we really want to check `$senderReferences` and not just `$config['transports']`.

Well, actually, below, there is not a check for the "global" transport, but we could add one if we follow my comment below about creating a `$failureTransportsByName` array.